crates/tx5/src/peer.rs

```diff
@@ -339,23 +360,5 @@ async fn task(
             .await;
     }
 
-    // the above loop ended, notify of disconnect
-    match tokio::time::timeout(
```

Comment:
By sending the disconnect here, it is sent before the peer is removed from the peer map.

Thus we may remove close the sbd connection for the peer and remove it from the hub map, but still have a Peer available that can be interacted with. Those interactions will fail.

Instead this change is to only send the disconnect when the Peer is dropped.